commit 82567fb0984625e5e11b89ecbfbbb4654a2a313b
Author: Takashi Sakamoto <o-takashi@sakamocchi.jp>
Date:   Mon Sep 17 17:26:08 2018 +0900

    ALSA: oxfw: fix memory leak for model-dependent data at error path
    
    commit ce925f088b979537f22f9e05eb923ef9822ca139 upstream.
    
    After allocating model-dependent data, ALSA OXFW driver has memory leak
    of the data at error path.
    
    This commit releases the data at the error path.
    
    Fixes: 6c29230e2a5f ('ALSA: oxfw: delayed registration of sound card')
    Cc: <stable@vger.kernel.org> # v4.7+
    Signed-off-by: Takashi Sakamoto <o-takashi@sakamocchi.jp>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/sound/firewire/oxfw/oxfw.c b/sound/firewire/oxfw/oxfw.c
index 1e5b2c802635..331cd5f65741 100644
--- a/sound/firewire/oxfw/oxfw.c
+++ b/sound/firewire/oxfw/oxfw.c
@@ -270,6 +270,8 @@ static void do_registration(struct work_struct *work)
 	if (oxfw->has_output)
 		snd_oxfw_stream_destroy_simplex(oxfw, &oxfw->tx_stream);
 	snd_card_free(oxfw->card);
+	kfree(oxfw->spec);
+	oxfw->spec = NULL;
 	dev_info(&oxfw->unit->device,
 		 "Sound card registration failed: %d\n", err);
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1036_82567fb0984625e5e11b89ecbfbbb4654a2a313b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1036_82567fb0984625e5e11b89ecbfbbb4654a2a313b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
